common/txmgr/confirmer.go

```diff
@@ -297,7 +304,20 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) pro
 		return fmt.Errorf("CheckConfirmedMissingReceipt failed: %w", err)
 	}
 
-	if err := ec.CheckForReceipts(ctx, head.BlockNumber()); err != nil {
+	_, latestFinalizedHead, err := ec.headTracker.LatestAndFinalizedBlock(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve latest finalized head: %w", err)
+	}
+
+	if !latestFinalizedHead.IsValid() {
+		return fmt.Errorf("latest finalized head is not valid")
+	}
+
+	if latestFinalizedHead.BlockNumber() > head.BlockNumber() {
```

Comment:
nit: Don't think this log is needed.
 The HeadTracker should never send finalzied blocks that are newer than current blocks.
This is likely a clear case of TXM panicking or throwing critical errors. But this should already been handled at HT layer.